src/slurm_plugin/clustermgtd.py

```diff
@@ -679,13 +751,42 @@ def _handle_powering_down_nodes(self, slurm_nodes):
                 instances_to_terminate, terminate_batch_size=self._config.terminate_max_batch_size
             )
 
+    @staticmethod
+    def _console_output_callback(output):
+        compute_logger.info(
+            "Console output for node %s (Instance Id %s):\r%s",
+            output.get("Name"),
+            output.get("InstanceId"),
+            output.get("ConsoleOutput"),
+        )
+
+    def _report_console_output_from_nodes(self, compute_nodes):
+        if self._config.compute_console_logging_enabled:
+            if self._config.compute_console_logging_max_sample_size > 0:
+                report_nodes = compute_nodes[: self._config.compute_console_logging_max_sample_size]
+            else:
+                report_nodes = compute_nodes
+
+            self._queue_executor_task(
+                self._instance_manager.get_console_output_task(
```

Comment:
Changed the queue function to raise an exception if the backlog is exceeded.
Also, the task themselves should log errors if they encounter an exception.
And finally, I change the task exception handlers to re-raise any encountered exception so that the exception my be retrieved from the returned future object.